ibis/expr/rules.py

```diff
@@ -134,11 +134,16 @@ def sort_key_from(table_ref, key, **kwargs):
     else:
         key, order = key, True
 
-    key = one_of(
-        (function_of(table_ref), column_from(table_ref), any),
-        key,
-        **kwargs,
-    )
+    if isinstance(key, (str, int)):
+        # Actual str/int keys must refer to columns in the table, we don't want
+        # to fallback to converting them to expressions with ibis.literal
+        key = column_from(table_ref, key, **kwargs)
+    else:
+        key = one_of(
+            (function_of(table_ref), column_from(table_ref), any),
```

Comment:
How about removing `any`?

Reply:
It's still legitimate to order by a constant, despite likely limited utility. IMO we shouldn't make that judgment call (yet).

Reply:
What is the point of ordering by a constant (at least in ibis land)? Ordering by a function like `RAND()` would make sense, but we can add a rule for that.

Reply:
Well, in one case, it's a workaround for snowflake's hard requirement for an `ORDER BY` in some window functions (even where techincally it isn't required).

I don't think we should make this choice on behalf of users right now. Maybe at some point we will, but it's also not causing any issues other than this one.

Reply:
I'm going to merge it, though it would be nice to better formalize sort key requirements in the future. It also applies to grouping keys and join keys which we are handling non uniformly.